HHS/Head-Start-TTADP
# [PROD] Legacy report updates, Other Entities filter, Formatted text paste fix, goal status visualization

# 456
## Description of change
Added goal status visualization to frontend (bar graph showing status/counts for a given recipients goals). It can be found on the goals & objectives tab of the recipient record page.

## How to test
View the goal status visualization. Confirm that it shows the counts returned from the backend and that filters apply to it. Confirm that the data can be viewing as a simple HTML table rather than a horizontal bar graph.

# 559
## Description of change
Some legacy reports are missing creator or collaborator data in the database despite having that data present in the smartsheet data column. This script attempts to assign collaborators from the "otherSpecialists" field and creators from the "createdBy" field.

## How to test
Run the script. You may not have data in your DB that works (emails in the "imported" data in my database don't match any of the users), so you might have to create some.

# 553
## Description of change

Created a script to alter the data some reports have in the delivery type column in the database, based on mappings in the jira.

## How to test
Imports from staging should have the offending reports. Locate them in your DB, run the script, and then confirm that those reports have been edited appropriately, and that the entire database wasn't mucked about with.

# 627
## Description of change

Added filter for Other Entities with type ahead.

## How to test

Go to the AR landing page and add a filter for Other Entities. AR's with this entity should only be shown.

# 644
## Description of change

See https://github.com/jpuri/react-draft-wysiwyg/issues/967. This change should make pasting formatted text behave better, most note-ably stopping random line breaks in bulleted lists.

before
![Screenshot from 2022-02-10 15-29-23](https://user-images.githubusercontent.com/6631790/153500495-425bca79-dec1-4bf1-a672-8359c1700aa7.png)

after
![Screenshot from 2022-02-10 15-29-36](https://user-images.githubusercontent.com/6631790/153500512-d7a112eb-0c0a-4489-ac20-e1d4cb80d2fb.png)

We will lose some formatting from the source, background highlights and font family. May be other things we lose. We should gather a list of all the formatting we want to carry over from word docs to verify what we want to work actually does work.

## How to test

From main, copy bulleted lists from a google doc into the context field. Note the extra newlines. Checkout this branch and copy the same text into the context field.

# 667
## Description of change
Updates the users in the seed used by processData for the non-production environments.

# 673
## Description of change
Fixes an error on selecting a filter with "Recipient name". 


## How to test
- On the landing page, create a new filter using "Recipient name contains"
- Verify that the Overview widget loads correctly (On Dev there is an error)
- Verify that the "Does not contain" clause also works

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-559
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-553
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-627
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-644
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-667
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-673


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
